src/ccnl-utils/src/ccn-lite-rpc.c

```diff
@@ -370,7 +370,7 @@ main(int argc, char *argv[])
             return -1;
         }
         nonce->type = LRPC_NONCE;
-        nonce->aux = malloc(sizeof(int));
+        nonce->aux = malloc(sizeof(struct rdr_ds_s));
         memcpy(nonce->aux, &n, sizeof(int));
```

Comment:
I am confused about this. 
There is a memcpy below opiing a random int to nonce->aux. Why is aux with size of int wrong?

Reply:
Because it is of type `` struct rdr_ds_s *aux;``? See also #324.

Reply:
> Or the other way around: why is this still initialized with a random int?

Good question. I can't really answer the question since I haven't designed anything ccn-lite RPC related nor wrote anything in ``ccn-lite-rpc.c``

Reply:
this inits the first int in the struct? right? 
Is that equal to nonce->aux->type = n?

Reply:
> this inits the first int in the struct? right?

I would have said that this initialises a nested element of type ``struct rdr_ds_s`` (a bit like a next pointer to the next element in a list).